comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -121,13 +151,26 @@ func (c *collector) GetTargetCatalog() workloadmeta.AgentType {
 func (c *collector) parseTasks(ctx context.Context, tasks []v1.Task) []workloadmeta.CollectorEvent {
 	events := []workloadmeta.CollectorEvent{}
 	seen := make(map[workloadmeta.EntityID]struct{})
+	// get task ARNs to fetch from the metadata v4 API
+	taskArns := c.getTaskArnsToFetch(tasks)
```

Comment:
Should we only get the task if `v4TaskEnabled` ?

Reply:
Good catch! It should be fix by 
https://github.com/DataDog/datadog-agent/blob/882d6e4354906f23c0124aa0d908eed0e5bcbd4a/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L202-L207

https://github.com/DataDog/datadog-agent/blob/882d6e4354906f23c0124aa0d908eed0e5bcbd4a/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L135-L139